src/lib.rs

```diff
@@ -77,6 +77,64 @@ use crate::r#impl::splice::make_splice_iterator;
 
 pub use crate::r#impl::{Drain, DrainFilter, IntoIter, Splice};
 
+#[inline]
+#[allow(clippy::cast_ptr_alignment)]
+fn is_default(buf: core::ptr::NonNull<u8>) -> bool {
+    core::ptr::eq(buf.as_ptr(), &DEFAULT_U8)
+}
```

Comment:
I don't understand the purpose of this refactor either.

Reply:
To use outside of generic context

Reply:
To try and reduce code bloat again?